File: code/day2.py
import fileHandle


def puzzle3(input_file):
    data = fileHandle.readfile(input_file).strip().splitlines()
    score = 0
    for round_ in data:
        hand1 = ord(round_[0]) - ord('A')
        hand2 = ord(round_[2]) - ord('X')
        score += hand2 + 1
        match hand2 - hand1:
            case 0:
                score += 3
            case 1 | -2:
                score += 6
    return score


def puzzle4(input_file):
    data = fileHandle.readfile(input_file).strip().splitlines()
    score = 0
    for round_ in data:
        hand1 = ord(round_[0]) - ord('A')
        result = ord(round_[2]) - ord('X')
        score += 3 * result
        match result:
            case 0:  # lose
                score += (hand1 - 1) % 3 + 1
            case 1:  # draw
                score += hand1 + 1
            case 2:  # win
                score += (hand1 + 1) % 3 + 1
    return score


print('Day #2, part one:', puzzle3('.\input\day2.txt'))
print('Day #2, part two:', puzzle4('.\input\day2.txt'))

File: code/day2.php
<?php
include 'file-handle.php';

function puzzle3($input_file)
{
    $data = read_file($input_file);
    $data = trim($data);
    $rounds = explode("\n", $data);
    $dict = [
        'A' => 1, 'B' => 2, 'C' => 3,
        'X' => 1, 'Y' => 2, 'Z' => 3,
    ];
    $total_score = 0;
    foreach ($rounds as $rnd) {
        $hand1 = $dict[$rnd[0]];    // $hand1 = ord($rnd[0]) - ord('A') + 1;
        $hand2 = $dict[$rnd[2]];    // $hand2 = ord($rnd[2]) - ord('X') + 1;
        $score = $hand2;
        switch ($hand2 - $hand1) {
            case 0: // draw
                $score += 3;
                break;
            case 1:
            case -2: // win
                $score += 6;
                break;
        }
        $total_score += $score;
    }
    return $total_score;
}


echo 'Day #2, part one: ' . puzzle3('./input/day2.txt') . PHP_EOL;
